rct229/rulesets/ashrae9012019/section19/section19rule35.py

```diff
@@ -0,0 +1,195 @@
+from rct229.rulesets.ashrae9012019.data.schema_enums import schema_enums
+from rct229.rulesets.ashrae9012019.ruleset_functions.aggregate_min_OA_schedule_across_zones import (
+    aggregate_min_OA_schedule_across_zones,
+)
+from rct229.rulesets.ashrae9012019.ruleset_functions.get_dict_of_zones_and_terminal_units_served_by_hvac_sys import (
+    get_dict_of_zones_and_terminal_units_served_by_hvac_sys,
+)
+from rct229.rulesets.ashrae9012019.ruleset_functions.get_min_oa_cfm_sch_zone import (
+    get_min_oa_cfm_sch_zone,
+)
+from rct229.utils.jsonpath_utils import find_all
+
+LIGHTING_SPACE = schema_enums["LightingSpaceOptions2019ASHRAE901TG37"]
+
+
+class Section19Rule35(RuleDefinitionListIndexedBase):
+    """Rule 35 of ASHRAE 90.1-2019 Appendix G Section 19 (HVAC - General)"""
+
+    def __init__(self):
+        super(Section19Rule35, self).__init__(
+            rmrs_used=UserBaselineProposedVals(False, True, False),
+            each_rule=Section19Rule35.HVACRule(),
+            index_rmr="baseline",
+            id="19-35",
+            description="For baseline systems serving only laboratory spaces that are prohibited from recirculating return air by code or accreditation standards, the baseline system shall be modeled as 100% outdoor air. Rule only applies when baseline outdoor air CFM is modeled as greater than proposed design outdoor air CFM.",
+            ruleset_section_title="HVAC - General",
+            standard_section="Section G3.1.2.5 Exception 4",
+            is_primary_rule=False,
+            rmr_context="ruleset_model_instances/0",
+            list_path="$.buildings[*].building_segments[*].heating_ventilating_air_conditioning_systems[*]",
+        )
+
+    def create_data(self, context, data):
+        rmi_b = context.baseline
+        rmi_p = context.proposed
+
+        dict_of_zones_and_terminal_units_served_by_hvac_sys_b = (
+            get_dict_of_zones_and_terminal_units_served_by_hvac_sys(rmi_b)
+        )
+        hvac_system_serves_only_labs = True
+        zone_OA_flow_list_of_schedules_b = []
+        zone_OA_flow_list_of_schedules_p = []
+        for zone_id_b in dict_of_zones_and_terminal_units_served_by_hvac_sys_b[
+            "hvac_id_b"
+        ]["zone_list"]:
+            for space_b in find_all(
+                f"$.buildings[*].building_segments[*].zones[?(@.id == {zone_id_b})].spaces[*]",
+                rmi_b,
+            ):
+                if hvac_system_serves_only_labs:
+                    if space_b.get("lighting_space_type") is not None:
+                        are_any_lighting_space_types_defined = True
+                        if (
+                            space_b.get("lighting_space_type")
+                            != LIGHTING_SPACE.LABORATORY_EXCEPT_IN_OR_AS_A_CLASSROOM
+                        ):
+                            hvac_system_serves_only_labs = False
+                    else:
+                        all_lighting_space_types_defined = False
```

Comment:
@weilixu good point! I think we can remove that part of the logic (line 53) and it can perform the subsequent logic on all spaces associated with the zone.

Reply:
Addressed.